src/SWP/Bundle/MultiTenancyBundle/DependencyInjection/Compiler/RegisterOrganizationFactoryPass.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher MultiTenancyBundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\MultiTenancyBundle\DependencyInjection\Compiler;
+
+use SWP\Component\MultiTenancy\Factory\OrganizationFactory;
+use SWP\Component\Storage\Factory\Factory;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Parameter;
+use Symfony\Component\DependencyInjection\Reference;
+
+class RegisterOrganizationFactoryPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
```

Comment:
This seems unnecessarily complicated to me for what you're trying achieve here. There is very little logic in this class, and I wonder if the same results could be achieved in yml. Should it actually be possible to create an organization factory with different parameters? Do we need that level of flexibility?